lib/internal/bootstrap/switches/does_own_process_state.js

```diff
@@ -139,6 +139,13 @@ function wrappedUmask(mask) {
 
 function wrappedCwd() {
   if (cachedCwd === '')
-    cachedCwd = rawMethods.cwd();
+    if (cachedCwd === '') {
+      try {
+        cachedCwd = rawMethods.cwd();
+      } catch (err) {
+        throw new Error(`process.cwd() failed: The current working directory was deleted.
+  Use process.chdir() to switch to a valid directory.`);
+      }
```

Comment:
Something like this would be better, we don't need to create a new error I think

```suggestion
    if (e.code === 'ENOENT' && e.syscall === 'uv_cwd') {
      // Enhance the error message
      e.message = 'Current working directory does not exist - this can happen if the directory ' +
                 'was deleted while the process was still inside it. Original error: ' + e.message;
    }
    throw e;
```